src/hotspot/share/opto/callGenerator.cpp

```diff
@@ -809,6 +813,514 @@ CallGenerator* CallGenerator::for_vector_reboxing_late_inline(ciMethod* method,
+
+    float probability_cache_exists = canonical_if_prob(_cache_not_null_iff);
+    hits_in_cache->set_profile_data(0, _cache_not_null_iff->_fcnt, probability_cache_exists);
+    float probability_first_cache_probe_fails = canonical_if_prob(_first_cache_probe_iff);
+    hits_in_cache->set_profile_data(1, if_cnt(_first_cache_probe_iff), probability_first_cache_probe_fails);
+    float probability_second_cache_probe_fails = canonical_if_prob(_second_cache_probe_iff);
+    hits_in_cache->set_profile_data(2, if_cnt(_second_cache_probe_iff), probability_second_cache_probe_fails);
+
+    Node* sv_hits_in_cachex = _kit->gvn().transform(hits_in_cache);
+    assert(sv_hits_in_cachex == hits_in_cache, "shouldn't be transformed to new node");
+
+    // And compute the probability of a miss in the cache
+    float cache_miss_prob;
+    // probability_cache_exists: probability that cache array is not null
+    // probability_first_cache_probe_fails: probability of a miss
+    // probability_second_cache_probe_fails: probability of a miss
+    if (probability_cache_exists == PROB_UNKNOWN || probability_first_cache_probe_fails == PROB_UNKNOWN || probability_second_cache_probe_fails == PROB_UNKNOWN) {
+      cache_miss_prob = PROB_UNKNOWN;
+    } else {
+      float probability_cache_does_not_exist = 1 - probability_cache_exists;
+      cache_miss_prob = probability_cache_does_not_exist + probability_cache_exists * probability_first_cache_probe_fails * probability_second_cache_probe_fails;
+    }
+
+    // Add the control flow that checks whether ScopedValueGetHitsInCache succeeds
+    Node* bol = _kit->gvn().transform(new BoolNode(hits_in_cache, BoolTest::ne));
+    IfNode* iff = new IfNode(_kit->control(), bol, 1 - cache_miss_prob, _cache_not_null_iff->_fcnt);
+    Node* transformed_iff = _kit->gvn().transform(iff);
+    assert(transformed_iff == iff, "shouldn't be transformed to new node");
+    Node* not_in_cache = _kit->gvn().transform(new IfFalseNode(iff));
+    Node* in_cache = _kit->gvn().transform(new IfTrueNode(iff));
+
+    // Merge the paths that produce the result (in case there's a slow path)
+    Node* region_fast_slow = new RegionNode(_slow_call == nullptr ? 2 : 3);
+    Node* phi_cache_value = new PhiNode(region_fast_slow, TypeInstPtr::BOTTOM);
+    Node* phi_mem = new PhiNode(region_fast_slow, Type::MEMORY, TypePtr::BOTTOM);
+    Node* phi_io = new PhiNode(region_fast_slow, Type::ABIO);
+
+    C->gvn_replace_by(scoped_value_cache_projs.fallthrough_catchproj, not_in_cache);
+    C->gvn_replace_by(scoped_value_cache_projs.resproj, scoped_value_cache_load);
+
+    if (_slow_call != nullptr) {
+      CallProjections slow_projs;
+      _slow_call->extract_projections(&slow_projs, false);
+      Node* fallthrough = slow_projs.fallthrough_catchproj->clone();
+      _kit->gvn().set_type(fallthrough, fallthrough->bottom_type());
+      region_fast_slow->init_req(2, fallthrough);
+      C->gvn_replace_by(slow_projs.fallthrough_catchproj, C->top());
```

Comment:
```suggestion
```
With this line moved up, the "region + 3 phi" are adjacent lines, just like you defined them above. Would be visually pleasing and easy to read.